kubernetes/utils/create_from_yaml.py

```diff
@@ -30,7 +30,7 @@ def create_from_yaml(
     Perform an action from a yaml file. Pass True for verbose to
     print confirmation information.
     Input:
-    yaml_file: string. Contains the path to yaml file.
+    yaml_file: string. Contains yaml string or a  path to yaml file.
```

Comment:
I may be a bit odd to expect a yaml string as input.
Wrapping preexisting strings into file-like objects is normally done by the caller the reasoning being if you have already have stream you have to read it into memory just for the function to convert it back to a memory stream.
Wrapping a pre-existing memory object into a stream on the otherhand is cheap as it just boils down to a memory view.

The usual interface is file path or file-like object. a file-like is also still acceptable for the argument name yaml_file, while that name does not really fit for a string.

Reply:
The reason we want to support string input is that ```kubectl create -f -``` takes a stdin input. I know it's kinda weird since bash does not really have dicts, but the functionality is there...

https://kubernetes.io/docs/reference/generated/kubectl/kubectl-commands#create

Reply:
that kind of my point stdin is not a string, stdin is a file-like object
so to use this function like `kubectl create -f -` you call `create_from_yaml(sys.stdin)`
the mapping of a potential `-` commandline argument to stdin should be done by the caller